### PR TITLE
fix: correctly report error in case of configureCustomCaCertificate failure

### PR DESCRIPTION
**Pull request by UtheMan**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_main.sh [hunk 1]
```diff
@@ -65,7 +65,7 @@ fi
 
 
 if [[ "${SHOULD_CONFIGURE_CUSTOM_CA_TRUST}" == "true" ]]; then
-    configureCustomCaCertificate || $ERR_UPDATE_CA_CERTS
```
[the remark this answers is not in the document]
Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/cse_main.sh:68]: not sure about how often this is actually failing,  was just perusing and this caught the corner of my eye

[the remark this answers is not in the document]
Comment (UtheMan) [reply parts/linux/cloud-init/artifacts/cse_main.sh:68]:
```bash
mkdir -p /opt/certs
for i in $(seq 0 $((${CUSTOM_CA_TRUST_COUNT} - 1))); do
    declare varname=CUSTOM_CA_CERT_${i} 
    echo "${!varname}" | base64 -d > /opt/certs/00000000000000cert${i}.crt
done
systemctl restart update_certs.service || exit $ERR_UPDATE_CA_CERTS
systemctl restart containerd
```

I suppose in this case, the main thing that can fail is the restart of update_certs.service. I can add logging + retries. Would retrying only `systemctl restart update_certs.service ` be good enough?

[the remark this answers is not in the document]
Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/cse_main.sh:68]: could the containerd restart also fail?

up to you, but I _think_ all that stuff is idempotent so might just be worth wrapping the original call in `retry_cmd_failure` or whatever it is

[the remark this answers is not in the document]
Comment (alexeldeib) [reply parts/linux/cloud-init/artifacts/cse_main.sh:68]: im okay to follow up and test this later, at least we'll get error viz now + data

**Review by alexeldeib — APPROVED** [review]

File: parts/linux/cloud-init/artifacts/cse_main.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (cameronmeissner) [reply parts/linux/cloud-init/artifacts/cse_main.sh:68]: also `logs_to_events`

**Review by cameronmeissner — APPROVED** [review]

**Pull request merged**